src/raftstore/store/snap.rs

```diff
@@ -748,17 +758,15 @@ impl Snapshot for Snap {
             "snapshot" => %self.path(),
         );
         for cf_file in &self.cf_files {
-            delete_file_if_exist(&cf_file.clone_path).unwrap();
-            if self.hold_tmp_files {
-                delete_file_if_exist(&cf_file.tmp_path).unwrap();
-            }
             if delete_file_if_exist(&cf_file.path).unwrap() {
+                // Here the file is used as a lock to avoid duplicated fetch_sub calls
+                // since delete can be called in parallel
                 self.size_track.fetch_sub(cf_file.size, Ordering::SeqCst);
             }
         }
-        delete_file_if_exist(&self.meta_file.path).unwrap();
+        delete_dir_if_exist(&self.subdir_path).unwrap();
```

Comment:
Can we only call delte_dir to delete the dir and files in this dir?

Reply:
See comments at L762